code/modules/clothing/shoes/magboots.dm

```diff
@@ -29,20 +29,31 @@
 		magpulse = 0
 		set_slowdown()
 		force = 3
-		if(icon_base) icon_state = "[icon_base]0"
 		to_chat(user, "You disable the mag-pulse traction system.")
 	else
 		item_flags |= ITEM_FLAG_NOSLIP
 		magpulse = 1
 		set_slowdown()
 		force = 5
-		if(icon_base) icon_state = "[icon_base]1"
 		playsound(get_turf(src), 'sound/effects/magnetclamp.ogg', 20)
 		to_chat(user, "You enable the mag-pulse traction system.")
-	user.update_inv_shoes()	//so our mob-overlays update
+	update_icon()
 	user.update_action_buttons()
 	user.update_floating()
 
+/obj/item/clothing/shoes/magboots/on_update_icon()
+	. = ..()
+	var/new_state = "[get_world_inventory_state()][magpulse]"
```

Comment:
Possibly `!!magpulse` to ensure it's 1/0?

Reply:
done